### DISCUSSION TITLE
Canonical tautomer

### DISCUSSION THREAD

**Discussion by apahl** [rdkit/rdkit#6374] [Q&A] [answered]
I know that tautomers are a controversial topic and I certainly do not want to start a big discussion, but I have just found some simple examples for canonical tautomers, which go against my chemical intuition and which are also in disagreement with other tools like MarvinSketch and Pipeline Pilot.

Here is one example: RDKit chooses as the canonical tautomer for 2-aminopyridine the exocyclic imine form, whereas the other tools (as well as my intuition) would keep the pyridine.
![image](https://github.com/rdkit/rdkit/assets/9406836/ba167061-fee9-4f97-86b5-8b9954ee4c76)

Any thoughts on this?
[notebook.zip](https://github.com/rdkit/rdkit/files/11487494/notebook.zip)

**Comment by bp-kelley** [comment]
I believe that the amino:imino ratio is around 1000:1 so I would be in agreement with you.

**Comment by kienerj** [comment]
I recently ran in unrelated issues as well an noticed that the max_tautomer setting can at times be "deceiving" and the default of 1000 should probably not be changed because it what you think should be the canonical tautomer doesn't appear in these 100, you get a different one. So I would try with 1000 and see if the issue persists.

Also I think there is a V1 and V2 version and V1 is probably like molvs and the new one slightly different?

But the core point is to get a canonical tautomer (eg always the same regardless which one you input). This might not be the "correct" one but it doesn't matter, it's more about getting a constant output. 

From documentation:

>Note that the canonical tautomer is very likely not the most stable tautomer for any given conditions. The default scoring rules are designed to produce “reasonable” tautomers, but the primary concern is that the results are canonical: you always get the same canonical tautomer for a molecule regardless of what the input tautomer or atom ordering were.

So this is more for comparing / matching and not actual "chemical relevance".

**Comment by greglandrum** [comment, marked as answer]
As @kienerj said: the primary goal of the canonical tautomer is to be canonical, not to reflect what is most stable in any given solvent at any particular temperature. We try to also have it be chemically "reasonable" but this isn't always going to work.

**Comment by apahl** [comment]
Thank you all for the valuable insights. I understand that the canonical tautomer does not necessarily represent the most stable one and that there are many cases where the differences between the tautomers are subtle and also that covering these cases with a rule-based approach is probably not possible. 
On the other hand, for the simple case that I have shown above (and I can give more of them, many involving amino-substituted heterocycles), I can not show the canonicalized structure to my boss for fear of being shouted at. It would be great, if the RDKit could give the more intuitive result here, I don't know if that is possible.
